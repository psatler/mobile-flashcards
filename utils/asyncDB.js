import { AsyncStorage } from 'react-native'
import { data } from './mockData'
const FLASHCARDS_DECKS_KEY = 'Flashcards:decks'

const setInitialData = async () => {
    let obj = data;

    await AsyncStorage.setItem(FLASHCARDS_DECKS_KEY, JSON.stringify(obj))
}


export const getDecks = async () => {
    try {
        let value = await AsyncStorage.getItem(FLASHCARDS_DECKS_KEY);
        if( value !== null){ //we have data
            // console.log('decks', value);
            let parsedObj = JSON.parse(value); //parsing back to object
            return parsedObj;
        } else { //if not, we set initial state
            console.log('setting initial state');
            setInitialData(); //defined above
        }
    } catch (err) {
        console.log('There was an error retrieving decks from local DB: ', err);
    }
}

export const saveDeckTitle = (keyTitle, deckObj) => {
    try {
        return AsyncStorage.mergeItem(FLASHCARDS_DECKS_KEY, JSON.stringify({
            [keyTitle]: deckObj
        }))
        
    } catch (err) {
        console.log(err);
    }
}

export const addCardToDeck = (deckTitle, card) => {
    const key = deckTitle;

    return AsyncStorage.getItem(FLASHCARDS_DECKS_KEY)
        .then( (results) => {
            const allDecks = JSON.parse(results);
            const deck = allDecks[key];
            // console.log('addCardToDeck- BEFORE', deck.questions)
            deck.questions.push(card)
            // console.log('addCardToDeck', deck.questions)
            // console.log('allDecks', allDecks)

            AsyncStorage.setItem(FLASHCARDS_DECKS_KEY, JSON.stringify(allDecks)); //updating 

        })
        // const key = deckTitle.split(' ').join('');

}

// 

// const checkIfDataExists = (results) => {
//     return results === null
//       ? setInitialData()
//       : JSON.parse(results)
//   }

// export const retrieveDataFromAsyncStorage = () => {

//     return AsyncStorage.getItem(FLASHCARDS_DECKS_KEY)
//         .then(checkIfDataExists)
// }



// _setValue = async () => {
//     try {
//       await AsyncStorage.setItem(STORAGE_KEY, 'true');
//     } catch (error) { // log the error
//     }
//   };

//   _loadInitialState = async () => {
//     try {
//       let value = await AsyncStorage.getItem(STORAGE_KEY);
//       if (value === 'true'){
//         this.setState({loaded: 'true'});
//       } else {
//         this.setState({loaded: 'false'});
//         this._setValue();
//       }
//     } catch (error) {
//       this.setState({loaded: 'false'});
//       this._setValue();
//     }
//   };