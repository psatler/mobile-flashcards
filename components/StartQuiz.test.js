import React from 'react'
import StartQuiz from './StartQuiz'
import { data } from '../utils/mockData'
import renderer from 'react-test-renderer';

//###### ENZYME STUFF
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })
import { shallow } from 'enzyme'


describe('[Component] StartQuiz', () => {
    let deckMock, navigationMock

    beforeEach( () => {
        //assigning tests
        deckMock = data['React'];
        navigationMock = { state: { params: { deck: deckMock } } } //passing one deck as prop (as it was coming from navigation.state.params.deck)
    })    

    it('shallow renders correctly', () => {
        const wrapper = shallow(<StartQuiz navigation={navigationMock} />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should toggle state property "isQuestion" to true or false', () => {
        const wrapper = renderer.create(<StartQuiz navigation={navigationMock} />).getInstance();
        expect(wrapper.state.isQuestion).toBeTruthy();
        wrapper.toggleQuestionAnswer();
        expect(wrapper.state.isQuestion).toBeFalsy();
        wrapper.toggleQuestionAnswer();
        expect(wrapper.state.isQuestion).toBeTruthy();
    });

    it('calls rightAnswer method', () => {
        const wrapper = renderer.create(<StartQuiz navigation={navigationMock} />).getInstance();
        expect(wrapper.state.currentIndex).toEqual(0);
        expect(wrapper.state.score).toEqual(0);
        wrapper.rightAnswer();
        expect(wrapper.state.currentIndex).toEqual(1);
        expect(wrapper.state.score).toEqual(1);
    });

    it('calls wrongAnswer method', () => {
        const wrapper = renderer.create(<StartQuiz navigation={navigationMock} />).getInstance();
        expect(wrapper.state.currentIndex).toEqual(0);
        expect(wrapper.state.score).toEqual(0);
        wrapper.wrongAnswer();
        expect(wrapper.state.currentIndex).toEqual(1);
        expect(wrapper.state.score).toEqual(0);
    });

    it('calls restartQuiz method', () => {
        const wrapper = renderer.create(<StartQuiz navigation={navigationMock} />).getInstance();
        wrapper.restartQuiz();
        expect(wrapper.state.isQuestion).toBeTruthy();
        expect(wrapper.state.currentIndex).toEqual(0);
        expect(wrapper.state.score).toEqual(0);
    });

    //############## Doing a mock below ###########
    it('should return there is no cards message', () => {
        deckMock = data['React'];
        deckMock.questions = []; //no cards
        navigationMock = { state: { params: { deck: deckMock } } }
        const wrapper = shallow(<StartQuiz navigation={navigationMock} />);
        expect(wrapper).toMatchSnapshot();
    });

    xit('should display the final result message', () => {
        // const wrapper = renderer.create(<StartQuiz navigation={navigationMock} />).getInstance(); //deck has one card
        // wrapper.rightAnswer(); // so the user answer correctly
        // expect(wrapper.toJSON()).toMatchSnapshot();

        const wrapper = shallow(<StartQuiz navigation={navigationMock} />);
        wrapper.setState({ currentIndex: 3 });
        expect(wrapper).toMatchSnapshot();
    })
    
})