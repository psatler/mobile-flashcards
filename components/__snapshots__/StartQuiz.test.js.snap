// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] StartQuiz shallow renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartQuiz
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "deck": Object {
              "questions": Array [
                Object {
                  "answer": "A library for managing user interfaces",
                  "question": "What is React?",
                },
                Object {
                  "answer": "The componentDidMount lifecycle event",
                  "question": "Where do you make Ajax requests in React?",
                },
              ],
              "title": "React",
            },
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flexDirection": "row",
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 0.5,
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 5,
              }
            }
          >
             
            1
             / 
            2
             
          </Text>
          <Component
            style={
              Object {
                "flex": 2,
              }
            }
          >
            <ProgressBar
              animated={true}
              animationConfig={
                Object {
                  "bounciness": 0,
                }
              }
              animationType="spring"
              borderRadius={4}
              borderWidth={1}
              color="#00A4D3"
              height={6}
              indeterminate={false}
              progress={0.5}
              style={
                Object {
                  "marginTop": 10,
                }
              }
              useNativeDriver={false}
              width={null}
            />
          </Component>
        </Component>,
        <Component
          style={
            Object {
              "flex": 1,
              "margin": 5,
              "marginBottom": 25,
            }
          }
        >
          <FlipCardAnimation
            answerText="A library for managing user interfaces"
            questionText="What is React?"
            showFront={true}
          />
        </Component>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#fff",
              "borderColor": "#4e4cb8",
              "borderRadius": 16,
              "borderWidth": 1,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#4e4cb8",
                "fontSize": 15,
              }
            }
          >
             See Answer 
          </Text>
        </TouchableOpacity>,
        <Component
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            disabled={true}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 16,
                  "marginBottom": 25,
                  "marginTop": 25,
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#757575",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 15,
                  "padding": 3,
                }
              }
            >
               Incorrect 
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            disabled={true}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 16,
                  "marginBottom": 25,
                  "marginTop": 25,
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#757575",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 15,
                  "padding": 3,
                }
              }
            >
               Correct 
            </Text>
          </TouchableOpacity>
        </Component>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 0.5,
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginTop": 5,
                }
              }
            >
               
              1
               / 
              2
               
            </Text>,
            <Component
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <ProgressBar
                animated={true}
                animationConfig={
                  Object {
                    "bounciness": 0,
                  }
                }
                animationType="spring"
                borderRadius={4}
                borderWidth={1}
                color="#00A4D3"
                height={6}
                indeterminate={false}
                progress={0.5}
                style={
                  Object {
                    "marginTop": 10,
                  }
                }
                useNativeDriver={false}
                width={null}
              />
            </Component>,
          ],
          "style": Object {
            "flexDirection": "row",
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 5,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                " ",
                1,
                " / ",
                2,
                " ",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "flex": 0.5,
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 5,
              },
            },
            "ref": null,
            "rendered": Array [
              " ",
              1,
              " / ",
              2,
              " ",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ProgressBar
                animated={true}
                animationConfig={
                  Object {
                    "bounciness": 0,
                  }
                }
                animationType="spring"
                borderRadius={4}
                borderWidth={1}
                color="#00A4D3"
                height={6}
                indeterminate={false}
                progress={0.5}
                style={
                  Object {
                    "marginTop": 10,
                  }
                }
                useNativeDriver={false}
                width={null}
              />,
              "style": Object {
                "flex": 2,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animated": true,
                "animationConfig": Object {
                  "bounciness": 0,
                },
                "animationType": "spring",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#00A4D3",
                "height": 6,
                "indeterminate": false,
                "progress": 0.5,
                "style": Object {
                  "marginTop": 10,
                },
                "useNativeDriver": false,
                "width": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FlipCardAnimation
            answerText="A library for managing user interfaces"
            questionText="What is React?"
            showFront={true}
          />,
          "style": Object {
            "flex": 1,
            "margin": 5,
            "marginBottom": 25,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "answerText": "A library for managing user interfaces",
            "questionText": "What is React?",
            "showFront": true,
          },
          "ref": "child",
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#4e4cb8",
                "fontSize": 15,
              }
            }
          >
             See Answer 
          </Text>,
          "onPress": [Function],
          "style": Object {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderColor": "#4e4cb8",
            "borderRadius": 16,
            "borderWidth": 1,
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": " See Answer ",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#4e4cb8",
              "fontSize": 15,
            },
          },
          "ref": null,
          "rendered": " See Answer ",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TouchableOpacity
              activeOpacity={0.2}
              disabled={true}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 16,
                    "marginBottom": 25,
                    "marginTop": 25,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "#757575",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "padding": 3,
                  }
                }
              >
                 Incorrect 
              </Text>
            </TouchableOpacity>,
            <TouchableOpacity
              activeOpacity={0.2}
              disabled={true}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 16,
                    "marginBottom": 25,
                    "marginTop": 25,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "#757575",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "padding": 3,
                  }
                }
              >
                 Correct 
              </Text>
            </TouchableOpacity>,
          ],
          "style": Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "padding": 3,
                  }
                }
              >
                 Incorrect 
              </Text>,
              "disabled": true,
              "onPress": [Function],
              "style": Array [
                Object {
                  "borderRadius": 16,
                  "marginBottom": 25,
                  "marginTop": 25,
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#757575",
                },
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": " Incorrect ",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#fff",
                  "fontSize": 15,
                  "padding": 3,
                },
              },
              "ref": null,
              "rendered": " Incorrect ",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "padding": 3,
                  }
                }
              >
                 Correct 
              </Text>,
              "disabled": true,
              "onPress": [Function],
              "style": Array [
                Object {
                  "borderRadius": 16,
                  "marginBottom": 25,
                  "marginTop": 25,
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#757575",
                },
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": " Correct ",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#fff",
                  "fontSize": 15,
                  "padding": 3,
                },
              },
              "ref": null,
              "rendered": " Correct ",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 0.5,
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginTop": 5,
                }
              }
            >
               
              1
               / 
              2
               
            </Text>
            <Component
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <ProgressBar
                animated={true}
                animationConfig={
                  Object {
                    "bounciness": 0,
                  }
                }
                animationType="spring"
                borderRadius={4}
                borderWidth={1}
                color="#00A4D3"
                height={6}
                indeterminate={false}
                progress={0.5}
                style={
                  Object {
                    "marginTop": 10,
                  }
                }
                useNativeDriver={false}
                width={null}
              />
            </Component>
          </Component>,
          <Component
            style={
              Object {
                "flex": 1,
                "margin": 5,
                "marginBottom": 25,
              }
            }
          >
            <FlipCardAnimation
              answerText="A library for managing user interfaces"
              questionText="What is React?"
              showFront={true}
            />
          </Component>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#fff",
                "borderColor": "#4e4cb8",
                "borderRadius": 16,
                "borderWidth": 1,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#4e4cb8",
                  "fontSize": 15,
                }
              }
            >
               See Answer 
            </Text>
          </TouchableOpacity>,
          <Component
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              disabled={true}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 16,
                    "marginBottom": 25,
                    "marginTop": 25,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "#757575",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "padding": 3,
                  }
                }
              >
                 Incorrect 
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              disabled={true}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "borderRadius": 16,
                    "marginBottom": 25,
                    "marginTop": 25,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "#757575",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "padding": 3,
                  }
                }
              >
                 Correct 
              </Text>
            </TouchableOpacity>
          </Component>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "flex": 0.5,
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "marginTop": 5,
                  }
                }
              >
                 
                1
                 / 
                2
                 
              </Text>,
              <Component
                style={
                  Object {
                    "flex": 2,
                  }
                }
              >
                <ProgressBar
                  animated={true}
                  animationConfig={
                    Object {
                      "bounciness": 0,
                    }
                  }
                  animationType="spring"
                  borderRadius={4}
                  borderWidth={1}
                  color="#00A4D3"
                  height={6}
                  indeterminate={false}
                  progress={0.5}
                  style={
                    Object {
                      "marginTop": 10,
                    }
                  }
                  useNativeDriver={false}
                  width={null}
                />
              </Component>,
            ],
            "style": Object {
              "flexDirection": "row",
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 5,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  " ",
                  1,
                  " / ",
                  2,
                  " ",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "flex": 0.5,
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginTop": 5,
                },
              },
              "ref": null,
              "rendered": Array [
                " ",
                1,
                " / ",
                2,
                " ",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ProgressBar
                  animated={true}
                  animationConfig={
                    Object {
                      "bounciness": 0,
                    }
                  }
                  animationType="spring"
                  borderRadius={4}
                  borderWidth={1}
                  color="#00A4D3"
                  height={6}
                  indeterminate={false}
                  progress={0.5}
                  style={
                    Object {
                      "marginTop": 10,
                    }
                  }
                  useNativeDriver={false}
                  width={null}
                />,
                "style": Object {
                  "flex": 2,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "animated": true,
                  "animationConfig": Object {
                    "bounciness": 0,
                  },
                  "animationType": "spring",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#00A4D3",
                  "height": 6,
                  "indeterminate": false,
                  "progress": 0.5,
                  "style": Object {
                    "marginTop": 10,
                  },
                  "useNativeDriver": false,
                  "width": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FlipCardAnimation
              answerText="A library for managing user interfaces"
              questionText="What is React?"
              showFront={true}
            />,
            "style": Object {
              "flex": 1,
              "margin": 5,
              "marginBottom": 25,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "answerText": "A library for managing user interfaces",
              "questionText": "What is React?",
              "showFront": true,
            },
            "ref": "child",
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#4e4cb8",
                  "fontSize": 15,
                }
              }
            >
               See Answer 
            </Text>,
            "onPress": [Function],
            "style": Object {
              "alignSelf": "center",
              "backgroundColor": "#fff",
              "borderColor": "#4e4cb8",
              "borderRadius": 16,
              "borderWidth": 1,
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": " See Answer ",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#4e4cb8",
                "fontSize": 15,
              },
            },
            "ref": null,
            "rendered": " See Answer ",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TouchableOpacity
                activeOpacity={0.2}
                disabled={true}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "borderRadius": 16,
                      "marginBottom": 25,
                      "marginTop": 25,
                      "padding": 10,
                    },
                    Object {
                      "backgroundColor": "#757575",
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 15,
                      "padding": 3,
                    }
                  }
                >
                   Incorrect 
                </Text>
              </TouchableOpacity>,
              <TouchableOpacity
                activeOpacity={0.2}
                disabled={true}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "borderRadius": 16,
                      "marginBottom": 25,
                      "marginTop": 25,
                      "padding": 10,
                    },
                    Object {
                      "backgroundColor": "#757575",
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 15,
                      "padding": 3,
                    }
                  }
                >
                   Correct 
                </Text>
              </TouchableOpacity>,
            ],
            "style": Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 15,
                      "padding": 3,
                    }
                  }
                >
                   Incorrect 
                </Text>,
                "disabled": true,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "borderRadius": 16,
                    "marginBottom": 25,
                    "marginTop": 25,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "#757575",
                  },
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": " Incorrect ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "padding": 3,
                  },
                },
                "ref": null,
                "rendered": " Incorrect ",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 15,
                      "padding": 3,
                    }
                  }
                >
                   Correct 
                </Text>,
                "disabled": true,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "borderRadius": 16,
                    "marginBottom": 25,
                    "marginTop": 25,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "#757575",
                  },
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": " Correct ",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "padding": 3,
                  },
                },
                "ref": null,
                "rendered": " Correct ",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`[Component] StartQuiz should display the final result message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartQuiz
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "deck": Object {
              "questions": Array [],
              "title": "React",
            },
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "restartFunc": [Function],
      "score": 0,
      "singleDeck": Object {
        "questions": Array [],
        "title": "React",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "restartFunc": [Function],
        "score": 0,
        "singleDeck": Object {
          "questions": Array [],
          "title": "React",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`[Component] StartQuiz should return there is no cards message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartQuiz
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "deck": Object {
              "questions": Array [],
              "title": "React",
            },
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "restartFunc": [Function],
      "score": 0,
      "singleDeck": Object {
        "questions": Array [],
        "title": "React",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "restartFunc": [Function],
        "score": 0,
        "singleDeck": Object {
          "questions": Array [],
          "title": "React",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
